### PR TITLE
Scissors - Stella 

### PR DESCRIPTION
**Pull request by stlashi**

### CODE REVIEW

File: swap_meet/decor.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from swap_meet.item import Item
+
+#Wave 5:
+class Decor(Item):
+    def __init__(self,condition = 0, age =0):
```
Comment (beccaelenzil) [inline swap_meet/decor.py:5]: Good use of super/inheritance.

File: swap_meet/vendor.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    def add(self,item):
+        self.inventory.append(item)
+        return item
+
+    def remove(self,item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
```
Comment (beccaelenzil) [inline swap_meet/vendor.py:20]: Consider using the pattern where you check the edge case first:

```suggestion
        if item not in self.inventory:
            return False
            
        self.inventory.remove(item)
        return item
```

File: swap_meet/vendor.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        for item in self.inventory:
+            if item.category == category:
+                category_list.append(item)
+        return category_list
+
+#Wave 3:
+    def swap_items(self,friend, my_item, their_item):
+        #if len(self.inventory) == 0 or len(friend.inventory) ==0:
+            #return False
+        if my_item in self.inventory and their_item in friend.inventory:
```
Comment (beccaelenzil) [inline swap_meet/vendor.py:35]: Consider refactoring so you check the condition where you return `False` first, and then move the main logic out of the conditional.

File: swap_meet/vendor.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            return True
+        else:
+            return False
+
+#Wave 4
+    def swap_first_item(self,friend):
+        if len(self.inventory) == 0 or len(friend.inventory) ==0:
+            return False    
+        else:
+            return self.swap_items(friend,self.inventory[0],friend.inventory[0])
```
Comment (beccaelenzil) [inline swap_meet/vendor.py:49]: You can move this line our of the `else` to enhance readability.

File: swap_meet/vendor.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+        best_condition = 0
+        category_items = self.get_by_category(category)
+        if len(category_items) != 0:
+            for item in category_items:
+                if item.condition > best_condition:
+                    best_item = item
+                    best_condition = item.condition
+        return best_item
+
+    def swap_best_by_category(self,other,my_priority,their_priority):
```
Comment (beccaelenzil) [inline swap_meet/vendor.py:64]: Wow! So efficient. Good use of the previous methods as helper functions.

File: swap_meet/vendor.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+        their_offer = other.get_best_by_category(my_priority)
+        my_offer = self.get_best_by_category(their_priority)
+        swap_result = self.swap_items(other,my_offer,their_offer)
+        return swap_result   
+    
+
+#Wave 7 - Optional Enhancement 
+    #Helper function - get item with the min age
```
Comment (beccaelenzil) [inline swap_meet/vendor.py:72]: Great work implementing this optional part.

File: tests/test_wave_07.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    item_a = Item(age=2)
+    item_b = Item(age=4)
+    tai = Vendor(inventory=[item_a, item_b]) 
+
+    item_d = Item(age=2)
+    item_e = Item(age=4)
+    jesse = Vendor(inventory=[item_d, item_e])   
+    result = tai.swap_by_newest(jesse)
+
+    assert result is True
```
Comment (beccaelenzil) [inline tests/test_wave_07.py:29]: Good work writing your own tests! It's good practice to add a few more assertions here to make sure that the swap actually happened. Check that `tai` now has `item_d` and `jesse` has `item_a` (did I get that right?). I would also add an edge case test. What happens when the vendors have no inventory? What happens when all the items in their inventory have the same age?

**Review by beccaelenzil — COMMENTED** [review]
Great work on your first OOP project! You've met the learning goals around reading (and writing!) tests and using composition and inheritance. Your code is logical and readable, and you've made great use of helper functions. I've left a few comments on small ways you could consider refactoring, and ideas on a few additional tests/assertions. Keep up the hard work!
